test/helpers/kubectl.go

```diff
@@ -596,6 +596,28 @@ func (kub *Kubectl) WaitforDeployReady(namespace, name string, timeout time.Dura
 		&TimeoutConfig{Timeout: timeout})
 }
 
+// WaitforDaemonSetReady waits for all required replicas of a daemonset to be
+// ready. Upon timeout an error is returned.
+func (kub *Kubectl) WaitforDaemonSetReady(namespace, name string, timeout time.Duration) error {
+	var specReplicas, currentReplicas int32
+
+	body := func() bool {
+		ds := apps_v1.DaemonSet{}
+		err := kub.Exec(fmt.Sprintf("%s -n %s get daemonset %s -o json", KubectlCmd, namespace, name)).Unmarshal(&ds)
```

Comment:
The `*CmdRes` returned by `kub.Exec` may be nil, which would result in a panic.

Reply:
Ah, okay, if we are guaranteed we will return a non-nil pointer, then this is fine!